modules/shinezBidAdapter.js

```diff
@@ -0,0 +1,430 @@
+    }
+
+    const skip = deepAccess(bid, `mediaTypes.video.skip`);
+    if (skip) {
+      bidObject.skip = skip;
+    }
+
+    const linearity = deepAccess(bid, `mediaTypes.video.linearity`);
+    if (linearity) {
+      bidObject.linearity = linearity;
+    }
+  }
+
+  return bidObject;
+}
+
+function isBanner(bid) {
+  return bid.mediaTypes && bid.mediaTypes.banner;
+}
+
+/**
+ * Generate params that are common between all bids
+ * @param {single bid object} generalObject
+ * @param {bidderRequest} bidderRequest
+ * @returns {object} the common params object
+ */
+function generateGeneralParams(generalObject, bidderRequest) {
+  const domain = window.location.hostname;
+  const {syncEnabled, filterSettings} = config.getConfig('userSync') || {};
+  const {bidderCode} = bidderRequest;
+  const generalBidParams = generalObject.params;
+  const timeout = config.getConfig('bidderTimeout');
+
+  // these params are snake_case instead of camelCase to allow backwards compatability on the server.
+  // in the future, these will be converted to camelCase to match our convention.
+  const generalParams = {
+    wrapper_type: 'prebidjs',
+    wrapper_vendor: '$$PREBID_GLOBAL$$',
+    wrapper_version: '$prebid.version$',
+    adapter_version: ADAPTER_VERSION,
+    auction_start: timestamp(),
+    publisher_id: generalBidParams.org,
+    publisher_name: domain,
+    site_domain: domain,
+    dnt: (navigator.doNotTrack == 'yes' || navigator.doNotTrack == '1' || navigator.msDoNotTrack == '1') ? 1 : 0,
+    device_type: getDeviceType(navigator.userAgent),
+    ua: navigator.userAgent,
+    session_id: getBidIdParameter('auctionId', generalObject),
+    tmax: timeout
+  }
+
+  const userIdsParam = getBidIdParameter('userId', generalObject);
+  if (userIdsParam) {
+    generalParams.userIds = JSON.stringify(userIdsParam);
+  }
+
+  const ortb2Metadata = bidderRequest.ortb2 || {};
+  if (ortb2Metadata.site) {
+    generalParams.site_metadata = JSON.stringify(ortb2Metadata.site);
+  }
+  if (ortb2Metadata.user) {
+    generalParams.user_metadata = JSON.stringify(ortb2Metadata.user);
+  }
+
+  if (syncEnabled) {
+    const allowedSyncMethod = getAllowedSyncMethod(filterSettings, bidderCode);
+    if (allowedSyncMethod) {
+      generalParams.cs_method = allowedSyncMethod;
+    }
+  }
+
+  if (bidderRequest.uspConsent) {
+    generalParams.us_privacy = bidderRequest.uspConsent;
+  }
+
+  if (bidderRequest && bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies) {
+    generalParams.gdpr = bidderRequest.gdprConsent.gdprApplies;
+    generalParams.gdpr_consent = bidderRequest.gdprConsent.consentString;
+  }
+
+  if (generalBidParams.ifa) {
+    generalParams.ifa = generalBidParams.ifa;
+  }
+
+  if (generalObject.schain) {
+    generalParams.schain = getSupplyChain(generalObject.schain);
+  }
+
+  if (bidderRequest && bidderRequest.refererInfo) {
+    generalParams.referrer = deepAccess(bidderRequest, 'refererInfo.referer');

```

Comment:
look for site.ref

Reply:
Hey @patmmccann, as i understand `ortb2.site.page` and `ortb2.site.ref` should be for Prebid 7 - right?